demos/message/README.md

```diff
@@ -0,0 +1,32 @@
+# Message
+
+In this demo we will show the message passing mechanism. Since messages are not sharable to other threads, this is a way for a thread to communicate with others privately.
+
+To explain this demo, we will first introduce the following:
+
+* `send_msg` (in `src/user/thread.c`)
+* `get_msg` (in `src/user/thread.c`)
+
+## `send_msg`
+
+In AMT, each thread owns a ring buffer to store pending messages. When this function is called, the kernel will check the corresponding buffer of the destination thread.
+
+If the buffer is not full, the message will be copied into the destination thread's buffer according to the `destination` and `message` we specified and wait for the thread to take it out.
+
+If the buffer is full, the kernel will refuse to send the message, and thus the return value of this function will be `false`.
+
+## `get_msg`
+
+When a thread calls this function, the kernel checks whether there is a message in the buffer.
+
+If the buffer is non-empty, the kernel will copy the information of the message to the variables we gave. So we can view the `content` and `src` variables to know the content and which thread sent this message. 
```

Comment:
This is good but note that it also frees a slot in the messages:
"to the variables we gave and remove it from the message buffer."